src/validators/duplicate_stops.rs

```diff
@@ -0,0 +1,91 @@
+use crate::validators::issues::*;
+use geo::algorithm::haversine_distance::HaversineDistance;
+use geo::Point;
+use itertools::Itertools;
+
+pub fn validate(gtfs: &gtfs_structures::Gtfs) -> Vec<Issue> {
+    let mut issues = vec![];
+
+    for (stop_a, stop_b) in gtfs.stops.values().tuple_combinations() {
```

Comment:
now you could do `gtfs.stops.values().tuple_combinations().filter(duplicate_stops).map(make_duplicate_stop).collect()` ;)

As those function are only used here, maybe they could accept a tuple instead of two parameters allowing to use them as shown in my example